### ISSUE TITLE
Notifications fall behind apps

### ISSUE THREAD

**Issue by frash23** [moetunes/Nextwm#18]
**Issue:** When selecting a window, it can be moved on top of a notification (I use Dunst)

**Steps to reproduce:** Run dunst and launch a notification with `notify-send`, select an un-selected window below the notification. The window will be moved on top of the notification, covering it up

**Suggested solution:** Add an "on top" property in apps.conf similar to `POPPED` that allows users to specify certain apps to always be on top of others.

**Comment by moetunes** [comment]
tl;dr snapwm can't do anything with an override_redirect window.

If you look at the source for dunst, in x.c line 1038, the override_redirect flag is set to true. This option was made available in X so an application can set itself to be ignored by a window manager.
Like all good window managers, snapwm ignores windows with override_redirect set to true.
I guess you have a large timeout set for the dunst window which is why it is shown for long enough for other windows to be selected and cover it, which is probably not how knopwob expected it to be used.
The simplest way around your issue is to add an option to dunst to not set the override_redirect flag to true and POPPED dunst in apps.conf cause snapwm just ignores it now.
HTH

**Comment by frash23** [comment]
Thanks for the hint!
I'll try adding an option to toggle override_redirect in dunst.

**Comment by moetunes** [comment]
From a quick check of the source you can remove line 1038 and remove the "CWOverrideRedirect | " from the start of line 1052.

**Comment by frash23** [comment]
Now dunst notifications are making snapwm crash. The only error message I can see is "Is another window manager running? Exiting", despite already having started up correctly.

**Comment by moetunes** [comment]
It was just a quick check of the source I did...
Try just removing line 1038 and leave line 1052 as supplied.

**Comment by moetunes** [comment]
I hope you've sorted this out.
Since it's not a window manager issue I'll close the issue.

**Comment by frash23** [comment]
Yep, no problem. I think I got it working for now, thanks for the info.
